crates/evm/src/execute.rs

```diff
@@ -287,7 +280,38 @@ where
     pub(crate) ctx: F::ExecutionCtx<'a>,
     pub(crate) parent: &'a SealedHeader<HeaderTy<N>>,
     pub(crate) assembler: Builder,
-    pub(crate) simulated: bool,
+}
+
+/// Conversions for executable transactions.
+pub trait ExecutorTx<B: BlockExecutor> {
+    /// The inner signed transaction.
+    type T;
+
+    /// Converts the transaction into [`ExecutableTx`].
+    fn as_executable<'a>(&'a self) -> impl ExecutableTx<B> + 'a
+    where
+        <<B as BlockExecutor>::Evm as Evm>::Tx: alloy_evm::FromRecoveredTx<Self::T>;
+
+    /// Converts the transaction into [`Recovered`].
+    fn into_recovered(self) -> Recovered<Self::T>;
+}
+
+impl<T, B: BlockExecutor> ExecutorTx<B> for Recovered<T>
+where
+    for<'a> Recovered<&'a T>: RecoveredTx<<B as BlockExecutor>::Transaction>,
+{
+    type T = T;
+
+    fn as_executable<'a>(&'a self) -> impl ExecutableTx<B> + 'a
+    where
+        <<B as BlockExecutor>::Evm as Evm>::Tx: alloy_evm::FromRecoveredTx<T>,
+    {
+        self.as_recovered_ref()
```

Comment:
I believe this is also impl for &Recovered
so we could avoid the extra address copy by returning `self`